daemon/cmd/state.go

```diff
@@ -318,39 +318,30 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState, endpoi
 		}
 	}
 
-	if option.Config.EnableIPSec {
-		// If IPsec is enabled we need to restore the host endpoint before any
-		// other endpoint, to ensure a dropless upgrade.
-		// This code can be removed in v1.15.
-		// This is necessary because we changed how the IPsec encapsulation is
-		// done. In older version, bpf_lxc would pass the outer destination IP
-		// via skb->cb to bpf_host which would write it to the outer header.
-		// In newer versions, the header is written by the kernel XFRM
-		// subsystem and bpf_host must therefore not write it. To allow for a
-		// smooth upgrade, bpf_host has been updated to handle both cases. But
-		// for that to succeed, it must be reloaded first, before the bpf_lxc
-		// programs stop writing the IP into skb->cb.
-		for _, ep := range state.restored {
-			// Cap the timeout used to wait for remote cluster synchronization
-			// to avoid blocking the agent startup, as this regeneration is
-			// performed synchronously.
-			endpointsRegenerator.CapTimeoutForSynchronousRegeneration()
-
-			if ep.IsHost() {
-				log.WithField(logfields.EndpointID, ep.ID).Info("Successfully restored endpoint. Scheduling regeneration")
-				if err := ep.RegenerateAfterRestore(endpointsRegenerator, d.bwManager, d.fetchK8sMetadataForEndpoint); err != nil {
-					log.WithField(logfields.EndpointID, ep.ID).WithError(err).Debug("error regenerating restored host endpoint")
-					epRegenerated <- false
-				} else {
-					epRegenerated <- true
-				}
-				break
-			}
+	for _, ep := range state.restored {
+		// Cap the timeout used to wait for remote cluster synchronization
+		// to avoid blocking the agent startup, as this regeneration is
+		// performed synchronously.
+		endpointsRegenerator.CapTimeoutForSynchronousRegeneration()
```

Comment:
The function comment for `CapTimeoutForSynchronousRegeneration` still says this is only used when ipsec is enabled.

Reply:
Hmm, that might explain the clustermesh failures. I'll lift it out of the loop,